packages/core/src/actions/waitForTransactionReceipt.ts

```diff
@@ -65,17 +76,34 @@ export async function waitForTransactionReceipt<
     )
     const txn = await action_getTransaction({ hash: receipt.transactionHash })
     const action_call = getAction(client, call, 'call')
-    const code = await action_call({
-      ...(txn as any),
-      gasPrice: txn.type !== 'eip1559' ? txn.gasPrice : undefined,
-      maxFeePerGas: txn.type === 'eip1559' ? txn.maxFeePerGas : undefined,
-      maxPriorityFeePerGas:
-        txn.type === 'eip1559' ? txn.maxPriorityFeePerGas : undefined,
-    })
-    const reason = code?.data
-      ? hexToString(`0x${code.data.substring(138)}`)
-      : 'unknown reason'
-    throw new Error(reason)
+    try {
+      const code = await action_call({
+        ...(txn as any),
+        gasPrice: txn.type !== 'eip1559' ? txn.gasPrice : undefined,
+        maxFeePerGas: txn.type === 'eip1559' ? txn.maxFeePerGas : undefined,
+        maxPriorityFeePerGas:
+          txn.type === 'eip1559' ? txn.maxPriorityFeePerGas : undefined,
+        data: txn.input ? txn.input : undefined,
+      })
+
+      const reason = code?.data
+        ? hexToString(`0x${code.data.substring(138)}`)
+        : 'unknown reason'
+      throw new Error(reason)
+    } catch (error) {
+      if (
+        contract &&
+        error instanceof CallExecutionError &&
+        error.cause instanceof ExecutionRevertedError &&
+        error.cause.cause instanceof RpcRequestError
+      ) {
+        throw new ContractFunctionRevertedError({
+          abi: contract.abi as Abi,
+          functionName: contract.functionName,
+          data: (error.cause.cause.cause as any).data,
```

Comment:
can typing be slightly improved here?

Reply:
Removed this code as suggested in commend below